minari/dataset/minari_storage.py

```diff
@@ -31,15 +37,47 @@ def __init__(self, data_path: PathLike):
             ValueError: if the specified path doesn't exist or doesn't contain any data.
 
         """
-        if not os.path.exists(data_path) or not os.path.isdir(data_path):
+        self._data_path: pathlib.Path = data_path
+        self._observation_space = observation_space
+        self._action_space = action_space
+
+    @classmethod
+    def read(cls, data_path: PathLike) -> MinariStorage:  # TODO: docs
+        data_path = pathlib.Path(data_path)
+        if not data_path.exists() or not data_path.is_dir():
             raise ValueError(f"The data path {data_path} doesn't exist")
-        file_path = os.path.join(str(data_path), "main_data.hdf5")
-        if not os.path.exists(file_path):
+        metadata_file_path = data_path.joinpath(_METADATA_FILE_NAME)
+        if not metadata_file_path.exists():
             raise ValueError(f"No data found in data path {data_path}")
-        self._file_path = file_path
-
-        self._observation_space = None
-        self._action_space = None
+        with open(metadata_file_path) as file:
+            metadata = json.load(file)
+
+        observation_space = None
+        action_space = None
+        if "observation_space" in metadata.keys():
+            serialized_space = metadata["observation_space"]
+            assert isinstance(serialized_space, str)
+            observation_space = deserialize_space(serialized_space)
+        if "action_space" in metadata.keys():
+            serialized_space = metadata["action_space"]
+            assert isinstance(serialized_space, str)
+            action_space = deserialize_space(serialized_space)
+
+        if action_space is None or observation_space is None:
```

Comment:
I don't think it is possible to have `env_spec` None as well as spaces, no? Because we need to know the space for data serialization/deserialization. Also in #137  there is a check that doesn't allow this case